L103-python-add-typing-to-sync-api.md

```diff
@@ -0,0 +1,79 @@
+Add Typing to gRPC Python Sync API
+----
+* Author(s): XuanWnag-Amos
+* Approver: gnossen
+* Status: Draft
+* Implemented in: Python
+* Last updated: 11/09/2022
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Add type annotation to Python Sync API.
+
+## Background
+
+[Type Hints](https://peps.python.org/pep-0484/#union-types) was introduced for a while now, we already added type annotation to python Async APIs since it has a high enough version floor. Now it's time to do the same for Sync APIs.
+
+This proposal plans to add type annotation to python Sync public APIs which allows users to build safer code by running static type analysis tools.
+
+
+### Related Proposals: 
+* Loosely related: https://github.com/lidizheng/proposal/blob/grpc-python-async-api/L58-python-async-api.md#introduce-typing-to-generated-code
+  * The proposal includes the part to introduce typing to gRPC Python Async APIs.
+
+
+## Proposal
+
+Add type annotation to the following Python APIs:
+* All APIs in `__init__.py` (the actuall list is too long, please refer to following PR for changes):
```

Comment:
This definition is a little bit too piecemeal. What we actual plan to type is _all public APIs_ in the `grpc` module provided by the `grpcio` PyPi package. This is listed out in the [`__all__` variable](https://github.com/grpc/grpc/blob/54dd7563c2d563bff74e4b558f2e985db4a01f2d/src/python/grpcio/grpc/__init__.py#L2100). No need to call out runtime protos specifically.

The scope should probably also include the generated code ([example](https://github.com/grpc/grpc/blob/master/examples/python/helloworld/helloworld_pb2_grpc.py)), ancillary packages, including grpcio-reflection, grpcio-channelz, etc. That doesn't necessarily have to be in the scope of _this_ gRFC, but since it takes 2 weeks to get a gRFC in, it might be prudent to do them all at the same time.

Reply:
I see, use `__all__` variable make more sense, I'll change to that.

Also agree that it's better to include generated code and other packages in this gRFC too, I'll add them.